
# MotrpacRatTraining6moWATData

<!-- badges: start -->

[![R-CMD-check](https://github.com/PNNL-Comp-Mass-Spec/MotrpacRatTraining6moWATData/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/PNNL-Comp-Mass-Spec/MotrpacRatTraining6moWATData/actions/workflows/R-CMD-check.yaml)

<!-- badges: end -->

Contains data and code necessary to reproduce analyses.

## Installation

``` r
devtools::install_github("PNNL-Comp-Mass-Spec/MotrpacRatTraining6moWATData")
```

## MotrpacRatTraining6moWAT

The <a
href="https://pnnl-comp-mass-spec.github.io/MotrpacRatTraining6moWAT/"
target="_blank">MotrpacRatTraining6moWAT</a> R package contains helper
functions for analysis and visualization of the data contained in
MotrpacRatTraining6moWATData.

## References

1.  MoTrPAC Study Group *et al.* Temporal dynamics of the multi-omic
    response to endurance exercise training across tissues. *bioRxiv*
    2022.09.21.508770 (2022)
    <https://doi.org/10.1101/2022.09.21.508770>.

2.  Gay N, Jean Beltran P, Amar D, Jimenez-Morales D, MoTrPAC Study
    Group (2023). *MotrpacRatTraining6moData: Data for analysis of the
    MoTrPAC endurance exercise training study in 6-month-old rats.*
    <https://motrpac.github.io/MotrpacRatTraining6moData/>,
    <https://github.com/MoTrPAC/MotrpacRatTraining6moData>.

3.  Dolgalev, I. (2022). msigdbr: MSigDB Gene Sets for Multiple
    Organisms in a Tidy Data Format. R package version 7.5.1,
    <https:://igordot.github.io/msigdbr>

4.  Horvath, S. (2011). Weighted Network Analysis. *Springer New York*.
    <https://doi.org/10.1007/978-1-4419-8819-5>

5.  Langfelder P and Horvath S, WGCNA: an R package for weighted
    correlation network analysis. *BMC Bioinformatics 2008*, 9:559
    <https://doi.org/10.1186/1471-2105-9-559>

6.  Law, C. W., Alhamdoosh, M., Su, S., Dong, X., Tian, L., Smyth, G.
    K., & Ritchie, M. E. (2016). RNA-seq analysis is easy as 1-2-3 with
    limma, Glimma and edgeR. *F1000Research, 5*, ISCB Comm J-1408.
    <https://doi.org/10.12688/f1000research.9005.3>

7.  Liberzon, A., Birger, C., Thorvaldsdóttir, H., Ghandi, M.,
    Mesirov, J. P., & Tamayo, P. (2015). The Molecular Signatures
    Database (MSigDB) hallmark gene set collection. *Cell systems*,
    *1*(6), 417–425. <https://doi.org/10.1016/j.cels.2015.12.004>

8.  Liu, R., Holik, A. Z., Su, S., Jansz, N., Chen, K., Leong, H. S.,
    Blewitt, M. E., Asselin-Labat, M. L., Smyth, G. K., & Ritchie, M. E.
    (2015). Why weight? Modelling sample and observational level
    variability improves power in RNA-seq analyses. *Nucleic acids
    research*, *43*(15), e97. <https://doi.org/10.1093/nar/gkv412>

9.  Peter Langfelder, Steve Horvath (2012). Fast R Functions for Robust
    Correlations and Hierarchical Clustering. *Journal of Statistical
    Software*, *46*(11), 1–17. <http://www.jstatsoft.org/v46/i11/>

10. Phipson, B., Lee, S., Majewski, I. J., Alexander, W. S., &
    Smyth, G. K. (2016). ROBUST HYPERPARAMETER ESTIMATION PROTECTS
    AGAINST HYPERVARIABLE GENES AND IMPROVES POWER TO DETECT
    DIFFERENTIAL EXPRESSION. *The annals of applied statistics*,
    *10*(2), 946–963. <https://doi.org/10.1214/16-AOAS920>

11. Smyth G. K. (2004). Linear models and empirical bayes methods for
    assessing differential expression in microarray experiments.
    *Statistical applications in genetics and molecular biology*, *3*,
    Article3. <https://doi.org/10.2202/1544-6115.1027>

12. Smyth, G. K., Michaud, J., and Scott, H. (2005). The use of
    within-array replicate spots for assessing differential expression
    in microarray experiments. *Bioinformatics 21*(9), 2067–2075.
    <http://bioinformatics.oxfordjournals.org/content/21/9/2067>
    Preprint with corrections:
    <http://www.statsci.org/smyth/pubs/dupcor.pdf>

13. Zhang, B., & Horvath, S. (2005). A general framework for weighted
    gene co-expression network analysis. *Statistical applications in
    genetics and molecular biology*, *4*, Article17.
    <https://doi.org/10.2202/1544-6115.1128>
