---
output: github_document
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# motrpacWATData

<!-- badges: start -->

<!-- badges: end -->

What is the purpose of this package?

## Installation

``` r
devtools::install_github("PNNL-Comp-Mass-Spec/motrpacWATData")
```

## References

1.  Dolgalev, I. (2022). msigdbr: MSigDB Gene Sets for Multiple Organisms in a Tidy Data Format. R package version 7.5.1, <https:://igordot.github.io/msigdbr>

2.  Horvath, S. (2011). Weighted Network Analysis. *Springer New York*. <https://doi.org/10.1007/978-1-4419-8819-5>

3.  Langfelder P and Horvath S, WGCNA: an R package for weighted correlation network analysis. *BMC Bioinformatics 2008*, 9:559 <https://doi.org/10.1186/1471-2105-9-559>

4.  Law, C. W., Alhamdoosh, M., Su, S., Dong, X., Tian, L., Smyth, G. K., & Ritchie, M. E. (2016). RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR. *F1000Research, 5*, ISCB Comm J-1408. <https://doi.org/10.12688/f1000research.9005.3>

5.  Liberzon, A., Birger, C., Thorvaldsd√≥ttir, H., Ghandi, M., Mesirov, J. P., & Tamayo, P. (2015). The Molecular Signatures Database (MSigDB) hallmark gene set collection. *Cell systems, 1*(6), 417--425. <https://doi.org/10.1016/j.cels.2015.12.004>

6.  Liu, R., Holik, A. Z., Su, S., Jansz, N., Chen, K., Leong, H. S., Blewitt, M. E., Asselin-Labat, M. L., Smyth, G. K., & Ritchie, M. E. (2015). Why weight? Modelling sample and observational level variability improves power in RNA-seq analyses. *Nucleic acids research*, *43*(15), e97. <https://doi.org/10.1093/nar/gkv412>

7.  Peter Langfelder, Steve Horvath (2012). Fast R Functions for Robust Correlations and Hierarchical Clustering. *Journal of Statistical Software, 46*(11), 1--17. <http://www.jstatsoft.org/v46/i11/>

8.  Phipson, B., Lee, S., Majewski, I. J., Alexander, W. S., & Smyth, G. K. (2016). ROBUST HYPERPARAMETER ESTIMATION PROTECTS AGAINST HYPERVARIABLE GENES AND IMPROVES POWER TO DETECT DIFFERENTIAL EXPRESSION. *The annals of applied statistics*, *10*(2), 946--963. <https://doi.org/10.1214/16-AOAS920>

9. Smyth G. K. (2004). Linear models and empirical bayes methods for assessing differential expression in microarray experiments. *Statistical applications in genetics and molecular biology*, *3*, Article3. <https://doi.org/10.2202/1544-6115.1027>

10. Smyth, G. K., Michaud, J., and Scott, H. (2005). The use of within-array replicate spots for assessing differential expression in microarray experiments. *Bioinformatics 21*(9), 2067--2075. <http://bioinformatics.oxfordjournals.org/content/21/9/2067> Preprint with corrections: <http://www.statsci.org/smyth/pubs/dupcor.pdf>

11. Zhang, B., & Horvath, S. (2005). A general framework for weighted gene co-expression network analysis. *Statistical applications in genetics and molecular biology, 4*, Article17. <https://doi.org/10.2202/1544-6115.1128>

