<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Gene Ontology FGSEA results — GO_FGSEA • MotrpacRatTraining6moWATData</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Gene Ontology FGSEA results — GO_FGSEA"><meta property="og:description" content="Fast Gene Set Enrichment Analysis (FGSEA) of transcriptomics and
  proteomics differential analysis results using Gene Ontology terms from the
  Molecular Signatures Database (MSigDB v7.5.1)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MotrpacRatTraining6moWATData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/ADIPOCYTE_SIZE_STATS.html">Statistical analysis of adipocyte diameter</a>
    </li>
    <li>
      <a href="../articles/ANALYTES_STATS.html">Statistical analyses of plasma clinical analytes</a>
    </li>
    <li>
      <a href="../articles/MAIN_FGSEA.html">FGSEA of Omics Datasets</a>
    </li>
    <li>
      <a href="../articles/PHENO_WAT_STATS.html">Statistical analyses of post - pre training differences</a>
    </li>
    <li>
      <a href="../articles/PHOSPHO_KSEA.html">Kinase-Substrate Enrichment Analysis</a>
    </li>
    <li>
      <a href="../articles/PROT_MITOCARTA_FGSEA.html">Proteomics MitoCarta FGSEA</a>
    </li>
    <li>
      <a href="../articles/WAT_DA.html">Differential Analysis of Omics Datasets</a>
    </li>
    <li>
      <a href="../articles/WAT_WGCNA.html">WGCNA of Omics Datasets</a>
    </li>
    <li>
      <a href="../articles/WGCNA_MODULE_ORA.html">WGCNA Module ORA</a>
    </li>
    <li>
      <a href="../articles/FGSEA_heatmaps.html">Heatmaps of FGSEA Results</a>
    </li>
    <li>
      <a href="../articles/KSEA_heatmaps.html">Heatmaps of Kinase-Substrate Enrichment Analysis Results</a>
    </li>
    <li>
      <a href="../articles/METAB_TAG_heatmap.html">Heatmap of total triacylglycerol concentrations</a>
    </li>
    <li>
      <a href="../articles/MitoCarta_FGSEA_heatmaps.html">Plots of MitoCarta 3.0 FGSEA Results</a>
    </li>
    <li>
      <a href="../articles/WGCNA_METAB_module_heatmaps.html">Heatmaps of Metabolites Grouped by WGCNA Module</a>
    </li>
    <li>
      <a href="../articles/WGCNA_correlation_heatmaps.html">Heatmaps of WGCNA Module Correlations</a>
    </li>
    <li>
      <a href="../articles/WGCNA_eigenfeature_plots.html">Plots of WGCNA eigenfeatures for each -ome</a>
    </li>
    <li>
      <a href="../articles/WGCNA_module_ORA.html">Plots of WGCNA Module Over-Representation Analysis Results</a>
    </li>
    <li>
      <a href="../articles/limma_diagnostics.html">LIMMA Diagnostic Plots</a>
    </li>
    <li>
      <a href="../articles/lipid_regulatory_proteins_heatmap.html">Heatmap of lipid regulatory proteins</a>
    </li>
    <li>
      <a href="../articles/plot_FGSEA_male_vs_female_summary.html">Plots of Top FGSEA Terms from Male vs. Female Comparisons</a>
    </li>
    <li>
      <a href="../articles/plot_Hk2.html">Plot of hexokinase 2 (Hk2) protein log2 relative abundance</a>
    </li>
    <li>
      <a href="../articles/plot_TAG_chain_length_vs_logFC.html">Plots of TAG chain length or number of double bonds vs. log2 fold-change</a>
    </li>
    <li>
      <a href="../articles/plot_UpSet.html">Differential analysis UpSet Plots</a>
    </li>
    <li>
      <a href="../articles/plot_adipocyte_size_distribution.html">Plots of Adipocyte Size Distributions</a>
    </li>
    <li>
      <a href="../articles/plot_analytes.html">Plots of plasma clinical analytes</a>
    </li>
    <li>
      <a href="../articles/plot_mito_reads_and_cardiolipin.html">Plots of % mitochondrial reads and cardiolipin</a>
    </li>
    <li>
      <a href="../articles/plot_mtDNA.html">Plot of Mitochondrial DNA (mtDNA)</a>
    </li>
    <li>
      <a href="../articles/plot_pct_fat_vs_leptin.html">Plot of Change in % Body Fat vs. Plasma Leptin</a>
    </li>
    <li>
      <a href="../articles/plot_pre_post.html">Plots of pre and post training values</a>
    </li>
    <li>
      <a href="../articles/plot_proteins.html">Plots of log2 relative abundances of various proteins</a>
    </li>
    <li>
      <a href="../articles/volcano_plots.html">Volcano Plots</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moWATData/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Gene Ontology FGSEA results</h1>
    <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moWATData/blob/HEAD/R/data.R" class="external-link"><code>R/data.R</code></a></small>
    <div class="hidden name"><code>GO_FGSEA.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Fast Gene Set Enrichment Analysis (FGSEA) of transcriptomics and
  proteomics differential analysis results using Gene Ontology terms from the
  Molecular Signatures Database (MSigDB v7.5.1).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">TRNSCRPT_FGSEA</span> <span class="co"># transcriptomics</span></span>
<span><span class="va">PROT_FGSEA</span>     <span class="co"># proteomics</span></span></code></pre></div>
    </div>

    <div id="format">
    <h2>Format</h2>
    <p>A named list of 3 <code>data.frame</code> objects:</p>
<p></p><dl><dt>"trained_vs_SED"</dt>
<dd><p>All trained timepoints compared to their
    sex-matched sedentary control group. Example: F_1W - F_SED. Total of 8
    contrasts.</p></dd>

    <dt>"MvF_SED"</dt>
<dd><p>Sedentary males compared to sedentary females. Only 1
    contrast.</p></dd>

    <dt>"MvF_exercise_response"</dt>
<dd><p>Comparisons of the male and female
    training responses. Example: (M_8W - M_SED) - (F_8W - F_SED). Total of 4
    contrasts.</p></dd>

  
</dl><p>Each <code>data.frame</code> contains the following 12 variables:</p>
<p></p><dl><dt>pathway</dt>
<dd><p>character; Gene Ontology pathway identifier.</p></dd>

    <dt>gs_subcat</dt>
<dd><p>character; GO:BF (Biological Processes), GO:MF
    (Molecular Functions), or GO:CC (Cellular Components). See
    <a href="http://geneontology.org/docs/ontology-documentation/" class="external-link">http://geneontology.org/docs/ontology-documentation/</a></p></dd>

    <dt>gs_description</dt>
<dd><p>character; Gene Ontology set description. See
    <code><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/reference/update_GO_names.html" class="external-link">update_GO_names</a></code> for details.</p></dd>

    <dt>pval</dt>
<dd><p>numeric; enrichment p-value.</p></dd>

    <dt>padj</dt>
<dd><p>numeric; BH-adjusted p-value. P-values are adjusted across
    all contrasts within each ontology (GO:BP, GO:CC, GO:MF).</p></dd>

    <dt>log2err</dt>
<dd><p>numeric; the expected error for the standard deviation of
    the p-value logarithm.</p></dd>

    <dt>ES</dt>
<dd><p>numeric; gene set enrichment score.</p></dd>

    <dt>NES</dt>
<dd><p>numeric; normalized enrichment score. Accounts for differences
    in gene set size. Calculated as the quotient of the ES to the absolute
    mean of the permutation enrichment scores that have the same sign as the
    ES. See <a href="https://doi.org/10.1101/060012" class="external-link">https://doi.org/10.1101/060012</a> for details. Values in [-1,
    1] are not interesting.</p></dd>

    <dt>size</dt>
<dd><p>integer; number of genes in each pathway after it has been
    filtered to those that were present in the differential analysis
    results.</p></dd>

    <dt>contrast</dt>
<dd><p>character; the comparison being made.</p></dd>

    <dt>leadingEdge</dt>
<dd><p>list; the Entrez gene IDs that drive the enrichment.
    If the set has a negative ES, the genes are arranged from most to least
    negative; otherwise, they are arranged from most to least positive. See
    <a href="http://software.broadinstitute.org/gsea/doc/GSEAUserGuideTEXT.htm#_Running_a_Leading" class="external-link">http://software.broadinstitute.org/gsea/doc/GSEAUserGuideTEXT.htm#_Running_a_Leading</a>.</p></dd>

    <dt>leadingEdge_genes</dt>
<dd><p>list; the <code>leadingEdge</code> after mapping
    Entrez genes to more human-readable gene symbols. Any genes that could
    not be mapped were discarded, so <code>lengths(leadingEdge_genes)</code> may
    not match <code>lengths(leadingEdge)</code>.</p></dd>

  
</dl></div>
    <div id="references">
    <h2>References</h2>
    <p>Ashburner, M., Ball, C. A., Blake, J. A., Botstein, D., Butler, H., Cherry,
  J. M., Davis, A. P., Dolinski, K., Dwight, S. S., Eppig, J. T., Harris, M.
  A., Hill, D. P., Issel-Tarver, L., Kasarskis, A., Lewis, S., Matese, J. C.,
  Richardson, J. E., Ringwald, M., Rubin, G. M., &amp; Sherlock, G. (2000). Gene
  ontology: tool for the unification of biology. The Gene Ontology
  Consortium. <em>Nature genetics, 25</em>(1), 25–29.
  <a href="https://doi.org/10.1038/75556" class="external-link">https://doi.org/10.1038/75556</a></p>
<p>Dolgalev, I. msigdbr: MSigDB Gene Sets for Multiple Organisms in a Tidy
  Data Format. R package version 7.5.1,
  <a href="https:://igordot.github.io/msigdbr">https:://igordot.github.io/msigdbr</a></p>
<p>Gene Ontology Consortium (2021). The Gene Ontology resource: enriching a
  GOld mine. <em>Nucleic acids research, 49</em>(D1), D325–D334.
  <a href="https://doi.org/10.1093/nar/gkaa1113" class="external-link">https://doi.org/10.1093/nar/gkaa1113</a></p>
<p>Korotkevich, G., Sukhov, V., Budin, N., Shpak, B., Artyomov, M. N., &amp;
  Sergushichev, A. (2021). Fast gene set enrichment analysis. <em>BioRxiv</em>.
  <a href="https://doi.org/10.1101/060012" class="external-link">https://doi.org/10.1101/060012</a></p>
<p>Liberzon, A., Birger, C., Thorvaldsdóttir, H., Ghandi, M.,
  Mesirov, J. P., &amp; Tamayo, P. (2015). The Molecular Signatures Database
  (MSigDB) hallmark gene set collection. <em>Cell systems, 1</em>(6), 417–425.
  <a href="https://doi.org/10.1016/j.cels.2015.12.004" class="external-link">https://doi.org/10.1016/j.cels.2015.12.004</a></p>
<p>Subramanian, A., Tamayo, P., Mootha, V. K., Mukherjee, S., Ebert, B. L.,
  Gillette, M. A., Paulovich, A., Pomeroy, S. L., Golub, T. R., Lander, E.
  S., &amp; Mesirov, J. P. (2005). Gene set enrichment analysis: A
  knowledge-based approach for interpreting genome-wide expression profiles.
  <em>Proceedings of the National Academy of Sciences of the United States
  of America, 102</em>(43), 15545–15550.
  <a href="https://doi.org/10.1073/pnas.0506580102" class="external-link">https://doi.org/10.1073/pnas.0506580102</a></p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="https://rdrr.io/pkg/fgsea/man/fgseaMultilevel.html" class="external-link">fgseaMultilevel</a></code>,
  <code><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/reference/fgsea2.html" class="external-link">fgsea2</a></code>,
  <code><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/reference/msigdbr2.html" class="external-link">msigdbr2</a></code>,
  <code><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/reference/enrichmat.html" class="external-link">enrichmat</a></code>, <code><a href="WAT_DA.html">PROT_DA</a></code>,
  <code><a href="WAT_DA.html">TRNSCRPT_DA</a></code>,
  <code><a href="msigdb_pathways.html">MSIGDB_PATHWAYS</a></code></p></div>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Tyler J. Sagendorf, Zhenxin Hou.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

