<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="An MSnSet-class object containing
transcriptomics data from rat WAT provided by MoTrPAC."><title>Transcriptomics MSnSet — TRNSCRPT_MSNSET • motrpacWATData</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Transcriptomics MSnSet — TRNSCRPT_MSNSET"><meta property="og:description" content="An MSnSet-class object containing
transcriptomics data from rat WAT provided by MoTrPAC."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">motrpacWATData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Transcriptomics MSnSet</h1>
      
      <div class="d-none name"><code>TRNSCRPT_MSNSET.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>An <code>MSnSet-class</code> object containing
transcriptomics data from rat WAT provided by MoTrPAC.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">TRNSCRPT_MSNSET</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>An <code>MSnSet-class</code> object with 16443 features and
48 samples.</p><div class="section">
<h3 id="assaydata"><code>assayData</code><a class="anchor" aria-label="anchor" href="#assaydata"></a></h3>


<p>A <code>matrix</code> of counts with 16443 rows and 48 columns.</p>
</div>

<div class="section">
<h3 id="featuredata"><code>featureData</code><a class="anchor" aria-label="anchor" href="#featuredata"></a></h3>


<p>A <code>data.frame</code> with 30304 rows and 7 variables containing additional
feature information:</p><dl><dt>feature_ID</dt>
<dd><p>character;</p></dd>

<dt>gene_symbol</dt>
<dd><p>character list; the gene symbol associated with each
protein.</p></dd>

<dt>entrez_gene</dt>
<dd><p>numeric list; the Entrez gene identifier associated
with each protein.</p></dd>

<dt>site</dt>
<dd><p>character;</p></dd>

<dt>human_feature_ID</dt>
<dd><p>character;</p></dd>

<dt>human_uniprot</dt>
<dd><p>character;</p></dd>

<dt>human_site</dt>
<dd><p>character;</p></dd>


</dl><p>Prior to collapsing the <code>entrez_id</code> and <code>gene_symbol</code> columns into lists,
any rows where <code>gene_symbol</code> began with "LOC" were removed unless that
would remove entire features. This reduces the impact of the one-to-many
feature-to-gene mapping on FGSEA results.</p>
</div>

<div class="section">
<h3 id="phenodata"><code>phenoData</code><a class="anchor" aria-label="anchor" href="#phenodata"></a></h3>


<p>A <code>data.frame</code> with 48 rows and 9 variables containing sample information:</p><dl><dt>viallabel</dt>
<dd><p>character; unique sample identifier.</p></dd>

<dt>rin</dt>
<dd><p>numeric; RNA integrity number (RIN).</p></dd>

<dt>pct_globin</dt>
<dd><p>numeric; percent of reads mapping to globin.</p></dd>

<dt>pct_umi_dup</dt>
<dd><p>numeric; percent of PCR duplicates as quantified with
Unique Molecular Identifiers (UMIs).</p></dd>

<dt>median_5_3_bias</dt>
<dd><p>numeric; median 5'-3' bias.</p></dd>

<dt>bid</dt>
<dd><p>character; Unique 5 digit numeric identifier of all samples
collected for an acute test/sample collection
period. All samples collected during that period will have the same BID.
Same as first 5 digits of <code>viallabel</code>.</p></dd>

<dt>sex</dt>
<dd><p>factor; the sex of the rat with 2 levels "Female" and "Male".</p></dd>

<dt>timepoint</dt>
<dd><p>factor; exercise training group. Either "SED"
(sedentary) or the number of weeks of training ("1W", "2W", "4W", "8W").</p></dd>

<dt>exp_group</dt>
<dd><p>factor; unique combination of <code>sex</code> (first letter) and
<code>timepoint</code>.</p></dd>


</dl></div>

    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Two outlier samples ("90423", "90410") were removed (48 remain). Then,
<code>pct_globin</code>, <code>rin</code>, <code>pct_umi_dup</code>, and <code>median_5_3_bias</code> were mean-imputed,
centered, and scaled.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">TRNSCRPT_MSNSET</span> <span class="co"># summary</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: MSnbase</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>there is no package called ‘MSnbase’</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in .requirePackage(package):</span> unable to find required package ‘MSnbase’</span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MSnbase</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in library(MSnbase):</span> there is no package called ‘MSnbase’</span>
<span class="r-in"><span><span class="fu">exprs</span><span class="op">(</span><span class="va">TRNSCRPT_MSNSET</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span> <span class="co"># assayData</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in exprs(TRNSCRPT_MSNSET):</span> could not find function "exprs"</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">fData</span><span class="op">(</span><span class="va">TRNSCRPT_MSNSET</span><span class="op">)</span><span class="op">)</span>     <span class="co"># featureData</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in fData(TRNSCRPT_MSNSET):</span> could not find function "fData"</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">pData</span><span class="op">(</span><span class="va">TRNSCRPT_MSNSET</span><span class="op">)</span><span class="op">)</span>     <span class="co"># phenoData</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in pData(TRNSCRPT_MSNSET):</span> could not find function "pData"</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Tyler J. Sagendorf, Zhenxin Hou.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

