<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="An object of class
  MSnbase::MSnSet containing
  transcriptomics data for differential analysis and (once transformed to
  log\(_2\) counts per million reads) WGCNA."><title>Transcriptomics MSnSet — TRNSCRPT_MSNSET • MotrpacRatTraining6moWATData</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Transcriptomics MSnSet — TRNSCRPT_MSNSET"><meta property="og:description" content="An object of class
  MSnbase::MSnSet containing
  transcriptomics data for differential analysis and (once transformed to
  log\(_2\) counts per million reads) WGCNA."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MotrpacRatTraining6moWATData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/PNNL-Comp-Mass-Spec/MotrpacRatTraining6moWATData/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Transcriptomics MSnSet</h1>
      <small class="dont-index">Source: <a href="https://github.com/PNNL-Comp-Mass-Spec/MotrpacRatTraining6moWATData/blob/HEAD/R/data.R" class="external-link"><code>R/data.R</code></a></small>
      <div class="d-none name"><code>TRNSCRPT_MSNSET.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>An object of class
  <code><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">MSnbase::MSnSet</a></code> containing
  transcriptomics data for differential analysis and (once transformed to
  log\(_2\) counts per million reads) WGCNA.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">TRNSCRPT_MSNSET</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>Object of class <code><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">MSnbase::MSnSet</a></code>
  with 16404 features and 48 samples.</p>
<p></p><dl><dt><b>assayData</b></dt>
<dd><p>Object of class <code>matrix</code> with 16404 rows and 48 columns containing
      RSEM gene counts.</p></dd>

    <dt><b>featureData</b></dt>
<dd><p>Object of class <code>data.frame</code> with 16404 rows and 3 variables
      containing additional feature information:</p>
<p></p><dl><dt><code>feature_ID</code></dt>
<dd><p>character; Reference Sequence (RefSeq)
         protein identifier.</p></dd>

         <dt><code>gene_symbol</code></dt>
<dd><p>character; official gene symbol.</p></dd>

         <dt><code>entrez_gene</code></dt>
<dd><p>integer; Entrez gene identifier.</p></dd>

      
</dl><p></p></dd>

    <dt><b>phenoData</b></dt>
<dd><p>Object of class <code>data.frame</code> with 48 rows and 13 variables
      containing additional sample information:</p>
<p></p><dl><dt><code>pid</code></dt>
<dd><p>integer; randomly generated 8-digit identifier used
        in linkage to phenotypic data. All samples from the same animal have
        the same PID.</p></dd>

        <dt><code>bid</code></dt>
<dd><p>integer; unique 5 digit identifier of all samples
        collected for an acute test/sample collection period. All samples
        collected during that period will have the same BID.</p></dd>

        <dt><code>labelid</code></dt>
<dd><p>integer; unique 11 digit specimen label
        identifier, originating at the collection site, that provides a link
        to specimen processing and is used for shipments to the
        biorepository. Same as <code>viallabel</code> only in instances where
        aliquots are not further processed at the biorepository.</p></dd>

        <dt><code>viallabel</code></dt>
<dd><p>character; unique 11 digit sample vial
        identifier. Starts with the <code>bid</code>.</p></dd>

        <dt><code>sex</code></dt>
<dd><p>factor; the sex of the rat with levels "Female" and
        "Male".</p></dd>

        <dt><code>timepoint</code></dt>
<dd><p>factor; exercise training group. Either "SED"
        (sedentary) or the number of weeks of training ("1W", "2W", "4W",
        "8W").</p></dd>

        <dt><code>exp_group</code></dt>
<dd><p>factor; experimental group. Unique
        combination of <code>sex</code> (first letter) and <code>timepoint</code>.</p></dd>

        <dt><code>rin</code></dt>
<dd><p>numeric; RNA integrity number (RIN).</p></dd>

        <dt><code>pct_globin</code></dt>
<dd><p>numeric; percent of reads mapping to
        globin.</p></dd>

        <dt><code>pct_umi_dup</code></dt>
<dd><p>numeric; percent of PCR duplicates as
        quantified with Unique Molecular Identifiers (UMIs).</p></dd>

        <dt><code>median_5_3_bias</code></dt>
<dd><p>numeric; median 5'-3' bias.</p></dd>

      
</dl><p></p></dd>

  
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The original 32883 transcripts were filtered with
  <code><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">filterByExpr</a></code> using <code>group =
  TRNSCRPT_MSNSET[["exp_group"]]</code>. Then, library sizes were TMM normalized
  with <code><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></code>. Two outlier samples (90423017005
  and 90410017005) were removed before variables <code>pct_globin</code>,
  <code>rin</code>, <code>pct_umi_dup</code>, and <code>median_5_3_bias</code> were
  mean-imputed, centered, and scaled.</p>
<p><a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/OUTLIERS.html" class="external-link">OUTLIERS</a> for sample outlier information.</p>
<p><a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/FEATURE_TO_GENE.html" class="external-link">FEATURE_TO_GENE</a> for details of
  feature-to-gene mapping.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="https://lgatto.github.io/MSnbase/reference/MSnSet-class.html" class="external-link">MSnSet-class</a></code>,
  <a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/TRNSCRPT_RAW_COUNTS.html" class="external-link">TRNSCRPT_WATSC_RAW_COUNTS</a>,
  <a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/FEATURE_TO_GENE.html" class="external-link">FEATURE_TO_GENE</a>, <a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a>,
  <a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">filterByExpr</a>, <a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://lgatto.github.io/MSnbase" class="external-link">MSnbase</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">TRNSCRPT_MSNSET</span>                   <span class="co"># summary</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MSnSet (storageMode: lockedEnvironment)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assayData: 16404 features, 48 samples </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   element names: exprs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> protocolData: none</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> phenoData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sampleNames: 90245017005 90248017005 ... 90227017005 (48 total)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   varLabels: pid bid ... median_5_3_bias (11 total)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   varMetadata: labelDescription</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> featureData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   featureNames: ENSRNOG00000000001 ENSRNOG00000000007 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ENSRNOG00000062306 (16404 total)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   fvarLabels: feature_ID gene_symbol entrez_gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   fvarMetadata: labelDescription</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> experimentData: use 'experimentData(object)'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Annotation:  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - - - Processing information - - -</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  MSnbase version: 2.24.2 </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">TRNSCRPT_MSNSET</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span> <span class="co"># assayData (first 10 rows and 5 columns)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    90245017005 90248017005 90252017005 90265017005 90266017005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000001          28          14          23          28          35</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000007           5          12           7           7           1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000008         110          53         114          88          87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000010          54          17           3           7          14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000012         106         153         158         103         209</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000017          34          54          61          37          64</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000021         721         321         326         367         436</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000024         493         773         674         644         847</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000033         512         323         403         377         411</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000034         490         279         425         301         447</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">TRNSCRPT_MSNSET</span><span class="op">)</span><span class="op">)</span>      <span class="co"># featureData</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            feature_ID    gene_symbol entrez_gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000001 ENSRNOG00000000001 AABR07013255.1        &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000007 ENSRNOG00000000007           Gad1       24379</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000008 ENSRNOG00000000008           Alx4      296511</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000010 ENSRNOG00000000010          Cbln1      498922</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000012 ENSRNOG00000000012          Tcf15      296272</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000017 ENSRNOG00000000017         Steap1      297738</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">TRNSCRPT_MSNSET</span><span class="op">)</span><span class="op">)</span>      <span class="co"># phenoData</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  pid   bid       labelid   viallabel    sex timepoint exp_group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90245017005 10043527 90245 4.458696e-313 90245017005 Female       SED     F_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90248017005 10043950 90248 4.458844e-313 90248017005 Female       SED     F_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90252017005 10044337 90252 4.459042e-313 90252017005 Female       SED     F_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90265017005 10046119 90265 4.459684e-313 90265017005 Female       SED     F_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90266017005 10046380 90266 4.459734e-313 90266017005 Female       SED     F_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90560017005 10700102 90560 4.474259e-313 90560017005 Female        1W      F_1W</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    rin pct_globin pct_umi_dup median_5_3_bias</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90245017005  0.1534937 -0.3394055   3.1014362      0.98444319</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90248017005 -2.0568155  1.0182164   0.3599009     -2.07005015</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90252017005 -2.5479953  0.3394055  -0.0376748     -1.03230562</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90265017005 -1.5656357 -0.3394055  -0.5626291      0.53410122</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90266017005 -0.3376861  0.3394055  -0.4034058     -0.05982804</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90560017005 -2.3024054  1.6970274  -0.2760272     -1.97214972</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Tyler J. Sagendorf, Zhenxin Hou.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

