<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Transcriptomics ExpressionSet — TRNSCRPT_EXP • MotrpacRatTraining6moWATData</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Transcriptomics ExpressionSet — TRNSCRPT_EXP"><meta property="og:description" content="An object of class
  Biobase::ExpressionSet
  containing transcriptomics data for differential analysis and (once
  transformed to log\(_2\) counts per million reads) WGCNA."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MotrpacRatTraining6moWATData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/ADIPOCYTE_SIZE_STATS.html">Statistical analysis of adipocyte diameter</a>
    </li>
    <li>
      <a href="../articles/ANALYTES_STATS.html">Statistical analyses of plasma clinical analytes</a>
    </li>
    <li>
      <a href="../articles/MAIN_FGSEA.html">FGSEA of Omics Datasets</a>
    </li>
    <li>
      <a href="../articles/PHENO_WAT_STATS.html">Statistical analyses of post - pre training differences</a>
    </li>
    <li>
      <a href="../articles/PHOSPHO_KSEA.html">Kinase-Substrate Enrichment Analysis</a>
    </li>
    <li>
      <a href="../articles/PROT_MITOCARTA_FGSEA.html">Proteomics MitoCarta FGSEA</a>
    </li>
    <li>
      <a href="../articles/WAT_DA.html">Differential Analysis of Omics Datasets</a>
    </li>
    <li>
      <a href="../articles/WAT_WGCNA.html">WGCNA of Omics Datasets</a>
    </li>
    <li>
      <a href="../articles/WGCNA_MODULE_ORA.html">WGCNA Module ORA</a>
    </li>
    <li>
      <a href="../articles/FGSEA_heatmaps.html">Heatmaps of FGSEA Results</a>
    </li>
    <li>
      <a href="../articles/KSEA_heatmaps.html">Heatmaps of Kinase-Substrate Enrichment Analysis Results</a>
    </li>
    <li>
      <a href="../articles/METAB_TAG_heatmap.html">Heatmap of total triacylglycerol concentrations</a>
    </li>
    <li>
      <a href="../articles/MitoCarta_FGSEA_heatmaps.html">Plots of MitoCarta 3.0 FGSEA Results</a>
    </li>
    <li>
      <a href="../articles/WGCNA_METAB_module_heatmaps.html">Heatmaps of Metabolites Grouped by WGCNA Module</a>
    </li>
    <li>
      <a href="../articles/WGCNA_correlation_heatmaps.html">Heatmaps of WGCNA Module Correlations</a>
    </li>
    <li>
      <a href="../articles/WGCNA_eigenfeature_plots.html">Plots of WGCNA eigenfeatures for each -ome</a>
    </li>
    <li>
      <a href="../articles/WGCNA_module_ORA.html">Plots of WGCNA Module Over-Representation Analysis Results</a>
    </li>
    <li>
      <a href="../articles/limma_diagnostics.html">LIMMA Diagnostic Plots</a>
    </li>
    <li>
      <a href="../articles/lipid_regulatory_proteins_heatmap.html">Heatmap of lipid regulatory proteins</a>
    </li>
    <li>
      <a href="../articles/plot_FGSEA_male_vs_female_summary.html">Plots of Top FGSEA Terms from Male vs. Female Comparisons</a>
    </li>
    <li>
      <a href="../articles/plot_Hk2.html">Plot of hexokinase 2 (Hk2) protein log2 relative abundance</a>
    </li>
    <li>
      <a href="../articles/plot_TAG_chain_length_vs_logFC.html">Plots of TAG chain length or number of double bonds vs. log2 fold-change</a>
    </li>
    <li>
      <a href="../articles/plot_UpSet.html">Differential analysis UpSet Plots</a>
    </li>
    <li>
      <a href="../articles/plot_adipocyte_size_distribution.html">Plots of Adipocyte Size Distributions</a>
    </li>
    <li>
      <a href="../articles/plot_analytes.html">Plots of plasma clinical analytes</a>
    </li>
    <li>
      <a href="../articles/plot_mito_reads_and_cardiolipin.html">Plots of % mitochondrial reads and cardiolipin</a>
    </li>
    <li>
      <a href="../articles/plot_mtDNA.html">Plot of Mitochondrial DNA (mtDNA)</a>
    </li>
    <li>
      <a href="../articles/plot_pct_fat_vs_leptin.html">Plot of Change in % Body Fat vs. Plasma Leptin</a>
    </li>
    <li>
      <a href="../articles/plot_pre_post.html">Plots of pre and post training values</a>
    </li>
    <li>
      <a href="../articles/plot_proteins.html">Plots of log2 relative abundances of various proteins</a>
    </li>
    <li>
      <a href="../articles/volcano_plots.html">Volcano Plots</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moWATData/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Transcriptomics ExpressionSet</h1>
    <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moWATData/blob/HEAD/R/data.R" class="external-link"><code>R/data.R</code></a></small>
    <div class="hidden name"><code>TRNSCRPT_EXP.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>An object of class
  <code><a href="https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html" class="external-link">Biobase::ExpressionSet</a></code>
  containing transcriptomics data for differential analysis and (once
  transformed to log\(_2\) counts per million reads) WGCNA.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">TRNSCRPT_EXP</span></span></code></pre></div>
    </div>

    <div id="format">
    <h2>Format</h2>
    <p>Object of class
  <code><a href="https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html" class="external-link">Biobase::ExpressionSet</a></code> with
  16404 features and 48 samples.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The original 32883 transcripts were filtered with
  <code><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">filterByExpr</a></code> using <code>group =
  TRNSCRPT_EXP[["exp_group"]]</code>. Then, library sizes were TMM normalized
  with <code><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></code>. Two outlier samples (90423017005
  and 90410017005) were removed before variables <code>pct_globin</code>,
  <code>rin</code>, <code>pct_umi_dup</code>, and <code>median_5_3_bias</code> were
  mean-imputed, centered, and scaled.</p>
<p>See <code><a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/OUTLIERS.html" class="external-link">OUTLIERS</a></code> for sample outlier
  information.</p>
<p>See <code><a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/FEATURE_TO_GENE.html" class="external-link">FEATURE_TO_GENE</a></code> for details of
  feature-to-gene mapping.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html" class="external-link">ExpressionSet-class</a></code>,
  <code><a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/TRNSCRPT_RAW_COUNTS.html" class="external-link">TRNSCRPT_WATSC_RAW_COUNTS</a></code>,
  <code><a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/FEATURE_TO_GENE.html" class="external-link">FEATURE_TO_GENE</a></code>,
  <code><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></code>, <code><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">filterByExpr</a></code>,
  <code><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biobase" class="external-link">Biobase</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">TRNSCRPT_EXP</span>                   <span class="co"># summary</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExpressionSet (storageMode: lockedEnvironment)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assayData: 16404 features, 48 samples </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   element names: exprs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> protocolData: none</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> phenoData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sampleNames: 90245017005 90248017005 ... 90227017005 (48 total)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   varLabels: pid bid ... median_5_3_bias (11 total)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   varMetadata: labelDescription</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> featureData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   featureNames: ENSRNOG00000000001 ENSRNOG00000000007 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ENSRNOG00000062306 (16404 total)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   fvarLabels: feature_ID gene_symbol entrez_gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   fvarMetadata: labelDescription</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> experimentData: use 'experimentData(object)'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Annotation:  </span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">TRNSCRPT_EXP</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span> <span class="co"># assayData (first 10 rows and 5 columns)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    90245017005 90248017005 90252017005 90265017005 90266017005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000001          28          14          23          28          35</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000007           5          12           7           7           1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000008         110          53         114          88          87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000010          54          17           3           7          14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000012         106         153         158         103         209</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000017          34          54          61          37          64</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000021         721         321         326         367         436</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000024         493         773         674         644         847</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000033         512         323         403         377         411</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000034         490         279         425         301         447</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">TRNSCRPT_EXP</span><span class="op">)</span><span class="op">)</span>      <span class="co"># featureData</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            feature_ID    gene_symbol entrez_gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000001 ENSRNOG00000000001 AABR07013255.1        &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000007 ENSRNOG00000000007           Gad1       24379</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000008 ENSRNOG00000000008           Alx4      296511</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000010 ENSRNOG00000000010          Cbln1      498922</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000012 ENSRNOG00000000012          Tcf15      296272</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSRNOG00000000017 ENSRNOG00000000017         Steap1      297738</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">TRNSCRPT_EXP</span><span class="op">)</span><span class="op">)</span>      <span class="co"># phenoData</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  pid   bid       labelid   viallabel    sex timepoint exp_group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90245017005 10043527 90245 4.458696e-313 90245017005 Female       SED     F_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90248017005 10043950 90248 4.458844e-313 90248017005 Female       SED     F_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90252017005 10044337 90252 4.459042e-313 90252017005 Female       SED     F_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90265017005 10046119 90265 4.459684e-313 90265017005 Female       SED     F_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90266017005 10046380 90266 4.459734e-313 90266017005 Female       SED     F_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90560017005 10700102 90560 4.474259e-313 90560017005 Female        1W      F_1W</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    rin pct_globin pct_umi_dup median_5_3_bias</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90245017005  0.1534937 -0.3394055   3.1014362      0.98444319</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90248017005 -2.0568155  1.0182164   0.3599009     -2.07005015</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90252017005 -2.5479953  0.3394055  -0.0376748     -1.03230562</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90265017005 -1.5656357 -0.3394055  -0.5626291      0.53410122</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90266017005 -0.3376861  0.3394055  -0.4034058     -0.05982804</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90560017005 -2.3024054  1.6970274  -0.2760272     -1.97214972</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Tyler J. Sagendorf, Zhenxin Hou.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

