<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Differential analysis of proteins, transcripts, phosphorylation
  sites, and metabolites measured in rat scWAT."><title>Differential analysis results — WATSC_DA • MotrpacRatTraining6moWATData</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Differential analysis results — WATSC_DA"><meta property="og:description" content="Differential analysis of proteins, transcripts, phosphorylation
  sites, and metabolites measured in rat scWAT."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MotrpacRatTraining6moWATData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/PNNL-Comp-Mass-Spec/MotrpacRatTraining6moWATData/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Differential analysis results</h1>
      <small class="dont-index">Source: <a href="https://github.com/PNNL-Comp-Mass-Spec/MotrpacRatTraining6moWATData/blob/HEAD/R/data.R" class="external-link"><code>R/data.R</code></a></small>
      <div class="d-none name"><code>WATSC_DA.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Differential analysis of proteins, transcripts, phosphorylation
  sites, and metabolites measured in rat scWAT.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">PROT_DA</span>     <span class="co"># proteomics</span></span>
<span><span class="va">PHOSPHO_DA</span>  <span class="co"># phosphoproteomics</span></span>
<span><span class="va">TRNSCRPT_DA</span> <span class="co"># transcriptomics</span></span>
<span><span class="va">METAB_DA</span>    <span class="co"># metabolomics</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>A named list of 3 <code>data.frame</code> objects:</p>
<p></p><dl><dt>"trained_vs_SED"</dt>
<dd><p>All trained timepoints compared to their
    sex-matched sedentary control group. Example: F_1W - F_SED. Total of 8
    contrasts.</p></dd>

    <dt>"MvF_SED"</dt>
<dd><p>Sedentary males compared to sedentary females. Only 1
    contrast.</p></dd>

    <dt>"MvF_exercise_response"</dt>
<dd><p>Comparisons of the male and female
    training responses. Example: (M_8W - M_SED) - (F_8W - F_SED). Total of 4
    contrasts.</p></dd>

  
</dl><p>In addition to all columns present in the <code>MSnbase::fData</code> tables,
  each <code>data.frame</code> contains the following 8 variables:</p>
<p></p><dl><dt><code>logFC</code></dt>
<dd><p>numeric; difference in the weighted mean log\(_2\)
    values of the groups specified in the <code>contrast</code> column.</p></dd>

    <dt><code>AveExpr</code></dt>
<dd><p>numeric; mean log\(_2\) relative abundance of
    each feature.</p></dd>

    <dt><code>t</code></dt>
<dd><p>numeric; <acronym>LIMMA</acronym> moderated t-statistic.</p></dd>

    <dt><code>P.Value</code></dt>
<dd><p>numeric; p-value.</p></dd>

    <dt><code>adj.P.Val</code></dt>
<dd><p>numeric; Benjamini-Hochberg adjusted p-value.
    P-values are adjusted across all contrasts.</p></dd>

    <dt><code>feature</code></dt>
<dd><p>character; uniquely defines each feature that was
    tested. Same as <code>feature_ID</code>.</p></dd>

    <dt><code>contrast</code></dt>
<dd><p>factor; the contrast(s) being tested.</p></dd>

    <dt><code>se</code></dt>
<dd><p>numeric; standard error (<acronym>SE</acronym>) of the
    <code>logFC</code> column.</p></dd>

  
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Differential analysis was performed with
  <code><a href="https://pnnl-comp-mass-spec.github.io/MotrpacRatTraining6moWAT/reference/limma_full.html" class="external-link">MotrpacRatTraining6moWAT::limma_full</a></code>
  on the <code>MSnSet</code> objects.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></code>,
  <code><a href="https://pnnl-comp-mass-spec.github.io/MotrpacRatTraining6moWAT/reference/limma_full.html" class="external-link">limma_full</a></code>,
  <code><a href="PROT_MSNSET.html">PROT_MSNSET</a></code>, <code><a href="PHOSPHO_MSNSET.html">PHOSPHO_MSNSET</a></code>,
  <code><a href="TRNSCRPT_MSNSET.html">TRNSCRPT_MSNSET</a></code>, <code><a href="METAB_MSNSET.html">METAB_MSNSET</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Number of differential features (-ome-wide FDR &lt; 0.05)</span></span></span>
<span class="r-in"><span><span class="co"># Convenience function</span></span></span>
<span class="r-in"><span><span class="va">f1</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>  <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">with</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">contrast</span>, <span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">f1</span><span class="op">(</span><span class="va">PROT_DA</span><span class="op">)</span>     <span class="co"># Proteins</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $trained_vs_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> contrast       FALSE TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_1W - F_SED  9875   89</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_2W - F_SED  9743  221</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_4W - F_SED  9825  139</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_8W - F_SED  9839  125</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_1W - M_SED  9811  153</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_2W - M_SED  9888   76</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_4W - M_SED  9393  571</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_8W - M_SED  9310  654</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $MvF_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> contrast        FALSE TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_SED - F_SED  5738 4226</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $MvF_exercise_response</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> contrast                          FALSE TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_1W - M_SED) - (F_1W - F_SED)  9943   21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_2W - M_SED) - (F_2W - F_SED)  9926   38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_4W - M_SED) - (F_4W - F_SED)  9824  140</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_8W - M_SED) - (F_8W - F_SED)  9943   21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="fu">f1</span><span class="op">(</span><span class="va">PHOSPHO_DA</span><span class="op">)</span>  <span class="co"># Phosphosites</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $trained_vs_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> contrast       FALSE  TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_1W - F_SED 30065   239</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_2W - F_SED 29952   352</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_4W - F_SED 30083   221</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_8W - F_SED 30147   157</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_1W - M_SED 30235    69</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_2W - M_SED 30166   138</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_4W - M_SED 30022   282</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_8W - M_SED 29360   944</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $MvF_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> contrast        FALSE  TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_SED - F_SED 24276  6028</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $MvF_exercise_response</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> contrast                          FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_1W - M_SED) - (F_1W - F_SED) 30304</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_2W - M_SED) - (F_2W - F_SED) 30304</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_4W - M_SED) - (F_4W - F_SED) 30304</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_8W - M_SED) - (F_8W - F_SED) 30304</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="fu">f1</span><span class="op">(</span><span class="va">TRNSCRPT_DA</span><span class="op">)</span> <span class="co"># Transcripts</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $trained_vs_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> contrast       FALSE  TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_1W - F_SED 16237   167</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_2W - F_SED 16345    59</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_4W - F_SED 16349    55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_8W - F_SED 16211   193</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_1W - M_SED 16248   156</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_2W - M_SED 15875   529</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_4W - M_SED 16391    13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_8W - M_SED 16314    90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $MvF_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> contrast        FALSE  TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_SED - F_SED  6068 10336</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $MvF_exercise_response</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> contrast                          FALSE  TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_1W - M_SED) - (F_1W - F_SED) 16270   134</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_2W - M_SED) - (F_2W - F_SED) 16314    90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_4W - M_SED) - (F_4W - F_SED) 16390    14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_8W - M_SED) - (F_8W - F_SED) 16361    43</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="fu">f1</span><span class="op">(</span><span class="va">METAB_DA</span><span class="op">)</span>    <span class="co"># Metabolites</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $trained_vs_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> contrast       FALSE TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_1W - F_SED   969   94</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_2W - F_SED   928  135</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_4W - F_SED  1043   20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   F_8W - F_SED   924  139</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_1W - M_SED   973   90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_2W - M_SED   902  161</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_4W - M_SED   909  154</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_8W - M_SED   852  211</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $MvF_SED</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> contrast        FALSE TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   M_SED - F_SED   448  615</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $MvF_exercise_response</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> contrast                          FALSE TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_1W - M_SED) - (F_1W - F_SED)  1027   36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_2W - M_SED) - (F_2W - F_SED)  1025   38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_4W - M_SED) - (F_4W - F_SED)  1007   56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (M_8W - M_SED) - (F_8W - F_SED)  1031   32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Tyler J. Sagendorf, Zhenxin Hou.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

