<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>R Package for WAT Companion Paper • WATManuscript2022</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="R Package for WAT Companion Paper">
<meta property="og:description" content="More about what it does (maybe more than one line)
    Use four spaces when indenting paragraphs within the Description.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">WATManuscript2022</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="watmanuscript2022">WATManuscript2022<a class="anchor" aria-label="anchor" href="#watmanuscript2022"></a>
</h1></div>
<!-- badges: start -->
<!-- badges: end -->
<p>What is the purpose of this package?</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"PNNL-Comp-Mass-Spec/WATManuscript2022"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ol style="list-style-type: decimal">
<li><p>Dolgalev, I. (2022). msigdbr: MSigDB Gene Sets for Multiple Organisms in a Tidy Data Format. R package version 7.5.1, <a href="https:://igordot.github.io/msigdbr" class="uri">https:://igordot.github.io/msigdbr</a></p></li>
<li><p>Horvath, S. (2011). Weighted Network Analysis. <em>Springer New York</em>. <a href="https://doi.org/10.1007/978-1-4419-8819-5" class="external-link uri">https://doi.org/10.1007/978-1-4419-8819-5</a></p></li>
<li><p>Langfelder P and Horvath S, WGCNA: an R package for weighted correlation network analysis. <em>BMC Bioinformatics 2008</em>, 9:559 <a href="https://doi.org/10.1186/1471-2105-9-559" class="external-link uri">https://doi.org/10.1186/1471-2105-9-559</a></p></li>
<li><p>Law, C. W., Alhamdoosh, M., Su, S., Dong, X., Tian, L., Smyth, G. K., &amp; Ritchie, M. E. (2016). RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR. <em>F1000Research, 5</em>, ISCB Comm J-1408. <a href="https://doi.org/10.12688/f1000research.9005.3" class="external-link uri">https://doi.org/10.12688/f1000research.9005.3</a></p></li>
<li><p>Liberzon, A., Birger, C., Thorvaldsdóttir, H., Ghandi, M., Mesirov, J. P., &amp; Tamayo, P. (2015). The Molecular Signatures Database (MSigDB) hallmark gene set collection. <em>Cell systems, 1</em>(6), 417–425. <a href="https://doi.org/10.1016/j.cels.2015.12.004" class="external-link uri">https://doi.org/10.1016/j.cels.2015.12.004</a></p></li>
<li><p>Liu, R., Holik, A. Z., Su, S., Jansz, N., Chen, K., Leong, H. S., Blewitt, M. E., Asselin-Labat, M. L., Smyth, G. K., &amp; Ritchie, M. E. (2015). Why weight? Modelling sample and observational level variability improves power in RNA-seq analyses. <em>Nucleic acids research</em>, <em>43</em>(15), e97. <a href="https://doi.org/10.1093/nar/gkv412" class="external-link uri">https://doi.org/10.1093/nar/gkv412</a></p></li>
<li><p>Peter Langfelder, Steve Horvath (2012). Fast R Functions for Robust Correlations and Hierarchical Clustering. <em>Journal of Statistical Software, 46</em>(11), 1–17. <a href="http://www.jstatsoft.org/v46/i11/" class="external-link uri">http://www.jstatsoft.org/v46/i11/</a></p></li>
<li><p>Phipson, B., Lee, S., Majewski, I. J., Alexander, W. S., &amp; Smyth, G. K. (2016). ROBUST HYPERPARAMETER ESTIMATION PROTECTS AGAINST HYPERVARIABLE GENES AND IMPROVES POWER TO DETECT DIFFERENTIAL EXPRESSION. <em>The annals of applied statistics</em>, <em>10</em>(2), 946–963. <a href="https://doi.org/10.1214/16-AOAS920" class="external-link uri">https://doi.org/10.1214/16-AOAS920</a></p></li>
<li><p>Smyth G. K. (2004). Linear models and empirical bayes methods for assessing differential expression in microarray experiments. <em>Statistical applications in genetics and molecular biology</em>, <em>3</em>, Article3. <a href="https://doi.org/10.2202/1544-6115.1027" class="external-link uri">https://doi.org/10.2202/1544-6115.1027</a></p></li>
<li><p>Smyth, G. K., Michaud, J., and Scott, H. (2005). The use of within-array replicate spots for assessing differential expression in microarray experiments. <em>Bioinformatics 21</em>(9), 2067–2075. <a href="http://bioinformatics.oxfordjournals.org/content/21/9/2067" class="external-link uri">http://bioinformatics.oxfordjournals.org/content/21/9/2067</a> Preprint with corrections: <a href="http://www.statsci.org/smyth/pubs/dupcor.pdf" class="external-link uri">http://www.statsci.org/smyth/pubs/dupcor.pdf</a></p></li>
<li><p>Zhang, B., &amp; Horvath, S. (2005). A general framework for weighted gene co-expression network analysis. <em>Statistical applications in genetics and molecular biology, 4</em>, Article17. <a href="https://doi.org/10.2202/1544-6115.1128" class="external-link uri">https://doi.org/10.2202/1544-6115.1128</a></p></li>
</ol>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing WATManuscript2022</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Tyler Sagendorf <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-1552-4870" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Tyler Sagendorf.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
