<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>FGSEA of Omics Datasets â€¢ MotrpacRatTraining6moWATData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="FGSEA of Omics Datasets">
<meta property="og:description" content="MotrpacRatTraining6moWATData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MotrpacRatTraining6moWATData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ADIPOCYTE_SIZE_STATS.html">Statistical analysis of adipocyte diameter</a>
    </li>
    <li>
      <a href="../articles/ANALYTES_STATS.html">Statistical analyses of plasma clinical analytes</a>
    </li>
    <li>
      <a href="../articles/MAIN_FGSEA.html">FGSEA of Omics Datasets</a>
    </li>
    <li>
      <a href="../articles/PHENO_WAT_STATS.html">Statistical analyses of post - pre training differences</a>
    </li>
    <li>
      <a href="../articles/PHOSPHO_KSEA.html">Kinase-Substrate Enrichment Analysis</a>
    </li>
    <li>
      <a href="../articles/PROT_MITOCARTA_FGSEA.html">Proteomics MitoCarta FGSEA</a>
    </li>
    <li>
      <a href="../articles/WAT_DA.html">Differential Analysis of Omics Datasets</a>
    </li>
    <li>
      <a href="../articles/WAT_WGCNA.html">WGCNA of Omics Datasets</a>
    </li>
    <li>
      <a href="../articles/WGCNA_MODULE_ORA.html">WGCNA Module ORA</a>
    </li>
    <li>
      <a href="../articles/FGSEA_heatmaps.html">Heatmaps of FGSEA Results</a>
    </li>
    <li>
      <a href="../articles/KSEA_heatmaps.html">Heatmaps of Kinase-Substrate Enrichment Analysis Results</a>
    </li>
    <li>
      <a href="../articles/METAB_TAG_heatmap.html">Heatmap of total triacylglycerol concentrations</a>
    </li>
    <li>
      <a href="../articles/MitoCarta_FGSEA_heatmaps.html">Plots of MitoCarta 3.0 FGSEA Results</a>
    </li>
    <li>
      <a href="../articles/WGCNA_METAB_module_heatmaps.html">Heatmaps of Metabolites Grouped by WGCNA Module</a>
    </li>
    <li>
      <a href="../articles/WGCNA_correlation_heatmaps.html">Heatmaps of WGCNA Module Correlations</a>
    </li>
    <li>
      <a href="../articles/WGCNA_eigenfeature_plots.html">Plots of WGCNA eigenfeatures for each -ome</a>
    </li>
    <li>
      <a href="../articles/WGCNA_module_ORA.html">Plots of WGCNA Module Over-Representation Analysis Results</a>
    </li>
    <li>
      <a href="../articles/limma_diagnostics.html">LIMMA Diagnostic Plots</a>
    </li>
    <li>
      <a href="../articles/lipid_regulatory_proteins_heatmap.html">Heatmap of lipid regulatory proteins</a>
    </li>
    <li>
      <a href="../articles/plot_FGSEA_male_vs_female_summary.html">Plots of Top FGSEA Terms from Male vs. Female Comparisons</a>
    </li>
    <li>
      <a href="../articles/plot_Hk2.html">Plot of hexokinase 2 (Hk2) protein log2 relative abundance</a>
    </li>
    <li>
      <a href="../articles/plot_TAG_chain_length_vs_logFC.html">Plots of TAG chain length or number of double bonds vs. log2 fold-change</a>
    </li>
    <li>
      <a href="../articles/plot_UpSet.html">Differential analysis UpSet Plots</a>
    </li>
    <li>
      <a href="../articles/plot_adipocyte_size_distribution.html">Plots of Adipocyte Size Distributions</a>
    </li>
    <li>
      <a href="../articles/plot_analytes.html">Plots of plasma clinical analytes</a>
    </li>
    <li>
      <a href="../articles/plot_mito_reads_and_cardiolipin.html">Plots of % mitochondrial reads and cardiolipin</a>
    </li>
    <li>
      <a href="../articles/plot_mtDNA.html">Plot of Mitochondrial DNA (mtDNA)</a>
    </li>
    <li>
      <a href="../articles/plot_pct_fat_vs_leptin.html">Plot of Change in % Body Fat vs. Plasma Leptin</a>
    </li>
    <li>
      <a href="../articles/plot_pre_post.html">Plots of pre and post training values</a>
    </li>
    <li>
      <a href="../articles/plot_proteins.html">Plots of log2 relative abundances of various proteins</a>
    </li>
    <li>
      <a href="../articles/volcano_plots.html">Volcano Plots</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moWATData/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>FGSEA of Omics Datasets</h1>
                        <h4 data-toc-skip class="author">Tyler
Sagendorf</h4>
            
            <h4 data-toc-skip class="date">05 March, 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moWATData/blob/HEAD/vignettes/MAIN_FGSEA.Rmd" class="external-link"><code>vignettes/MAIN_FGSEA.Rmd</code></a></small>
      <div class="hidden name"><code>MAIN_FGSEA.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://motrpac.github.io/MotrpacRatTraining6moWATData/" class="external-link">MotrpacRatTraining6moWATData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/" class="external-link">MotrpacRatTraining6moWAT</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/" class="external-link">purrr</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>Fast Gene Set Enrichment Analysis (FGSEA)<span class="citation"><sup>[<a href="#ref-korotkevich_fgsea_2019">1</a>]</sup></span> will be performed
using Entrez gene identifiers for proteomics and transcriptomics and
RefMet metabolite identifiers for metabolomics. Gene Ontology gene sets
will be used for proteomics and phosphoproteomics, while RefMet chemical
subclasses will be used to group metabolites into metabolite sets. We
assume that the majority of the white adipose tissue (WAT) proteome and
transcriptome were captured by our datasets, so we only keep gene sets
that retain the majority of their genes after filtering to what was
present in our -omics datasets. That is, only those gene sets with at
least 85% of their original genes will be used for testing. Gene sets
will be filtered further based on size so that they must contain at
least 15 genes. For metabolomics, a minimum of 10 metabolites per
chemical subclass are required, though a membership filter will not be
applied. Sets contain no more than 300 members due to pre-filtering of
<code>MSIGDB_PATHWAYS</code> (see <code><a href="../reference/msigdb_pathways.html">help("MSIGDB_PATHWAYS")</a></code>
for details).</p>
<div class="section level3">
<h3 id="transcriptomics">Transcriptomics<a class="anchor" aria-label="anchor" href="#transcriptomics"></a>
</h3>
<p>There is a one-to-many mapping between transcripts and Entrez genes,
so we will need to account for this.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">";"</span>, <span class="va">TRNSCRPT_DA</span><span class="op">$</span><span class="va">MvF_SED</span><span class="op">$</span><span class="va">entrez_gene</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt; 16227   177</span></span>
<span><span class="co"># 177 transcripts have multiple genes. Separate to multiple rows</span></span>
<span></span>
<span><span class="va">TRNSCRPT_DA_SEP</span> <span class="op">&lt;-</span> <span class="va">TRNSCRPT_DA</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span>.f <span class="op">=</span> <span class="op">~</span> <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate_rows.html" class="external-link">separate_rows</a></span><span class="op">(</span><span class="va">.x</span>, <span class="va">entrez_gene</span>, <span class="va">gene_symbol</span>, sep <span class="op">=</span> <span class="st">";"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">entrez_gene</span> <span class="op">!=</span> <span class="st">"NA"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Entrez to gene symbol conversion vector for leading edge</span></span>
<span><span class="va">TRNSCRPT_entrez_to_symbol</span> <span class="op">&lt;-</span> <span class="va">TRNSCRPT_DA_SEP</span><span class="op">$</span><span class="va">MvF_SED</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">entrez_gene</span>, <span class="va">gene_symbol</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/enframe.html" class="external-link">deframe</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">TRNSCRPT_entrez_to_symbol</span><span class="op">)</span></span>
<span><span class="co">#&gt;    24379   296511   498922   296272   297738   362454 </span></span>
<span><span class="co">#&gt;   "Gad1"   "Alx4"  "Cbln1"  "Tcf15" "Steap1"  "Hebp1"</span></span>
<span></span>
<span><span class="co"># Select gene sets that are largely unchanged when filtering</span></span>
<span><span class="co"># to only those Entrez IDs present in the results.</span></span>
<span><span class="va">TRNSCRPT_MSIGDB</span> <span class="op">&lt;-</span> <span class="va">MSIGDB_PATHWAYS</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>entrez_gene <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">entrez_gene</span>, .f <span class="op">=</span> <span class="va">intersect</span>,</span>
<span>                           y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">TRNSCRPT_entrez_to_symbol</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         set_size_post <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lengths.html" class="external-link">lengths</a></span><span class="op">(</span><span class="va">entrez_gene</span><span class="op">)</span>,</span>
<span>         ratio <span class="op">=</span> <span class="va">set_size_post</span> <span class="op">/</span> <span class="va">set_size</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ratio</span> <span class="op">&gt;=</span> <span class="fl">0.85</span>, <span class="co"># at least 85% of the original set remains</span></span>
<span>         <span class="va">set_size_post</span> <span class="op">&gt;=</span> <span class="fl">15</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">TRNSCRPT_MSIGDB</span><span class="op">$</span><span class="va">gs_subcat</span><span class="op">)</span> <span class="co"># how many gene sets remain?</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; GO:BP GO:CC GO:MF </span></span>
<span><span class="co">#&gt;  2715   320   480</span></span>
<span><span class="co"># GO:BP GO:CC GO:MF</span></span>
<span><span class="co">#  2864   347   498</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># FGSEA</span></span>
<span><span class="va">TRNSCRPT_FGSEA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">TRNSCRPT_DA_SEP</span>, <span class="kw">function</span><span class="op">(</span><span class="va">res_i</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/reference/fgsea2.html" class="external-link">fgsea2</a></span><span class="op">(</span>pathways <span class="op">=</span> <span class="va">TRNSCRPT_MSIGDB</span>,</span>
<span>         stats <span class="op">=</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/reference/rank_genes.html" class="external-link">rank_genes</a></span><span class="op">(</span><span class="va">res_i</span>, genes <span class="op">=</span> <span class="st">"entrez_gene"</span><span class="op">)</span>,</span>
<span>         seed <span class="op">=</span> <span class="fl">0</span>, nPermSimple <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>         adjust.globally <span class="op">=</span> <span class="cn">TRUE</span>, nproc <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="co"># Map Entrez IDs in leading edge subset to gene symbols</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>leadingEdge_genes <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>      .x <span class="op">=</span> <span class="va">leadingEdge</span>,</span>
<span>      .f <span class="op">=</span> <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="va">TRNSCRPT_entrez_to_symbol</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span>    <span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="co"># Reorder columns</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">pathway</span>, <span class="va">gs_subcat</span>, <span class="va">gs_description</span>, <span class="fu"><a href="https://tidyselect.r-lib.org/reference/everything.html" class="external-link">everything</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/relocate.html" class="external-link">relocate</a></span><span class="op">(</span><span class="va">contrast</span>, .before <span class="op">=</span> <span class="va">leadingEdge</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="proteomics">Proteomics<a class="anchor" aria-label="anchor" href="#proteomics"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Entrez to gene symbol conversion vector for leading edge</span></span>
<span><span class="va">PROT_entrez_to_symbol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">PROT_EXP</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">entrez_gene</span>, <span class="va">gene_symbol</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/enframe.html" class="external-link">deframe</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">PROT_entrez_to_symbol</span><span class="op">)</span></span>
<span><span class="co">#&gt;     26194     26195     26198     26196     26204     26199 </span></span>
<span><span class="co">#&gt;  "Mt-nd2"  "Mt-co1"  "Mt-co2" "Mt-atp8"  "Mt-co3"  "Mt-nd3"</span></span>
<span></span>
<span><span class="co"># Select gene sets that are largely unchanged when filtering</span></span>
<span><span class="co"># to only those Entrez IDs present in the results.</span></span>
<span><span class="va">PROT_MSIGDB</span> <span class="op">&lt;-</span> <span class="va">MSIGDB_PATHWAYS</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>entrez_gene <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">entrez_gene</span>, .f <span class="op">=</span> <span class="va">intersect</span>,</span>
<span>                           y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">PROT_entrez_to_symbol</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         set_size_post <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lengths.html" class="external-link">lengths</a></span><span class="op">(</span><span class="va">entrez_gene</span><span class="op">)</span>,</span>
<span>         ratio <span class="op">=</span> <span class="va">set_size_post</span> <span class="op">/</span> <span class="va">set_size</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ratio</span> <span class="op">&gt;=</span> <span class="fl">0.85</span>, <span class="co"># at least 85% of the original set remains</span></span>
<span>         <span class="va">set_size_post</span> <span class="op">&gt;=</span> <span class="fl">15</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">PROT_MSIGDB</span><span class="op">$</span><span class="va">gs_subcat</span><span class="op">)</span> <span class="co"># how many gene sets remain?</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; GO:BP GO:CC GO:MF </span></span>
<span><span class="co">#&gt;   230    97    77</span></span>
<span><span class="co"># GO:BP GO:CC GO:MF</span></span>
<span><span class="co">#   234    99    78</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># FGSEA</span></span>
<span><span class="va">PROT_FGSEA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">PROT_DA</span>, <span class="kw">function</span><span class="op">(</span><span class="va">res_i</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/reference/fgsea2.html" class="external-link">fgsea2</a></span><span class="op">(</span>pathways <span class="op">=</span> <span class="va">PROT_MSIGDB</span>,</span>
<span>         stats <span class="op">=</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/reference/rank_genes.html" class="external-link">rank_genes</a></span><span class="op">(</span><span class="va">res_i</span>, genes <span class="op">=</span> <span class="st">"entrez_gene"</span><span class="op">)</span>,</span>
<span>         seed <span class="op">=</span> <span class="fl">0</span>, nPermSimple <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>         adjust.globally <span class="op">=</span> <span class="cn">TRUE</span>, nproc <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="co"># Map Entrez IDs in leading edge subset to gene symbols</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>leadingEdge_genes <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>      .x <span class="op">=</span> <span class="va">leadingEdge</span>,</span>
<span>      .f <span class="op">=</span> <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="va">PROT_entrez_to_symbol</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span>    <span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="co"># Reorder columns</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">pathway</span>, <span class="va">gs_subcat</span>, <span class="va">gs_description</span>, <span class="fu"><a href="https://tidyselect.r-lib.org/reference/everything.html" class="external-link">everything</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/relocate.html" class="external-link">relocate</a></span><span class="op">(</span><span class="va">contrast</span>, .before <span class="op">=</span> <span class="va">leadingEdge</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="metabolomics">Metabolomics<a class="anchor" aria-label="anchor" href="#metabolomics"></a>
</h3>
<p>Unlike with proteomics and transcriptomics, we are not limited to
testing terms that are largely unchanged after filtering. This is
because the RefMet chemical subclasses are homogenous groups (e.g.,
subsetting the acyl carnitines will still result in a group of only acyl
carnitines).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Reformat fData for use with fgsea2</span></span>
<span><span class="va">REFMET_SUBCLASSES</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">METAB_EXP</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">refmet_sub_class</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>feature <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">feature_ID</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>gs_subcat <span class="op">=</span> <span class="st">"refmet_sub_class"</span>,</span>
<span>         gs_exact_source <span class="op">=</span> <span class="va">refmet_sub_class</span>,</span>
<span>         gs_description <span class="op">=</span> <span class="va">refmet_sub_class</span>,</span>
<span>         set_size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lengths.html" class="external-link">lengths</a></span><span class="op">(</span><span class="va">feature</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">set_size</span> <span class="op">&gt;=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">REFMET_SUBCLASSES</span><span class="op">)</span> <span class="co"># 19</span></span>
<span><span class="co">#&gt; [1] 19</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># FGSEA</span></span>
<span><span class="va">METAB_FGSEA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">METAB_DA</span>, <span class="kw">function</span><span class="op">(</span><span class="va">res_i</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/reference/fgsea2.html" class="external-link">fgsea2</a></span><span class="op">(</span>pathways <span class="op">=</span> <span class="va">REFMET_SUBCLASSES</span>,</span>
<span>         gene_column <span class="op">=</span> <span class="st">"feature"</span>,</span>
<span>         stats <span class="op">=</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/reference/rank_genes.html" class="external-link">rank_genes</a></span><span class="op">(</span><span class="va">res_i</span>, genes <span class="op">=</span> <span class="st">"feature"</span><span class="op">)</span>,</span>
<span>         seed <span class="op">=</span> <span class="fl">0</span>, nPermSimple <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>         adjust.globally <span class="op">=</span> <span class="cn">TRUE</span>, nproc <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="co"># Reorder columns</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>refmet_sub_class <span class="op">=</span> <span class="va">pathway</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">starts_with</a></span><span class="op">(</span><span class="st">"gs_"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/relocate.html" class="external-link">relocate</a></span><span class="op">(</span><span class="va">contrast</span>, .before <span class="op">=</span> <span class="va">leadingEdge</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Save</span></span>
<span><span class="fu">usethis</span><span class="fu">::</span><span class="fu">use_data</span><span class="op">(</span><span class="va">TRNSCRPT_FGSEA</span>, internal <span class="op">=</span> <span class="cn">FALSE</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  version <span class="op">=</span> <span class="fl">3</span>, compress <span class="op">=</span> <span class="st">"bzip2"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">usethis</span><span class="fu">::</span><span class="fu">use_data</span><span class="op">(</span><span class="va">PROT_FGSEA</span>, internal <span class="op">=</span> <span class="cn">FALSE</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  version <span class="op">=</span> <span class="fl">3</span>, compress <span class="op">=</span> <span class="st">"bzip2"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">usethis</span><span class="fu">::</span><span class="fu">use_data</span><span class="op">(</span><span class="va">METAB_FGSEA</span>, internal <span class="op">=</span> <span class="cn">FALSE</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  version <span class="op">=</span> <span class="fl">3</span>, compress <span class="op">=</span> <span class="st">"bzip2"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.3 (2024-02-29)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] purrr_1.0.2                        tibble_3.2.1                      </span></span>
<span><span class="co">#&gt; [3] tidyr_1.3.1                        dplyr_1.1.4                       </span></span>
<span><span class="co">#&gt; [5] MotrpacRatTraining6moWAT_1.0.0     Biobase_2.62.0                    </span></span>
<span><span class="co">#&gt; [7] BiocGenerics_0.48.1                MotrpacRatTraining6moWATData_2.0.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3      rstudioapi_0.15.0       jsonlite_1.8.8         </span></span>
<span><span class="co">#&gt;   [4] shape_1.4.6.1           magrittr_2.0.3          ggbeeswarm_0.7.2       </span></span>
<span><span class="co">#&gt;   [7] rmarkdown_2.25          GlobalOptions_0.1.2     fs_1.6.3               </span></span>
<span><span class="co">#&gt;  [10] zlibbioc_1.48.0         ragg_1.2.7              vctrs_0.6.5            </span></span>
<span><span class="co">#&gt;  [13] memoise_2.0.1           RCurl_1.98-1.14         base64enc_0.1-3        </span></span>
<span><span class="co">#&gt;  [16] rstatix_0.7.2           htmltools_0.5.7         dynamicTreeCut_1.63-1  </span></span>
<span><span class="co">#&gt;  [19] curl_5.2.1              broom_1.0.5             Formula_1.2-5          </span></span>
<span><span class="co">#&gt;  [22] sass_0.4.8              bslib_0.6.1             htmlwidgets_1.6.4      </span></span>
<span><span class="co">#&gt;  [25] desc_1.4.3              impute_1.76.0           cachem_1.0.8           </span></span>
<span><span class="co">#&gt;  [28] lifecycle_1.0.4         iterators_1.0.14        pkgconfig_2.0.3        </span></span>
<span><span class="co">#&gt;  [31] Matrix_1.6-5            R6_2.5.1                fastmap_1.1.1          </span></span>
<span><span class="co">#&gt;  [34] GenomeInfoDbData_1.2.11 clue_0.3-65             digest_0.6.34          </span></span>
<span><span class="co">#&gt;  [37] colorspace_2.1-0        patchwork_1.2.0         AnnotationDbi_1.64.1   </span></span>
<span><span class="co">#&gt;  [40] S4Vectors_0.40.2        textshaping_0.3.7       Hmisc_5.1-1            </span></span>
<span><span class="co">#&gt;  [43] RSQLite_2.3.5           ggpubr_0.6.0            filelock_1.0.3         </span></span>
<span><span class="co">#&gt;  [46] latex2exp_0.9.6         fansi_1.0.6             httr_1.4.7             </span></span>
<span><span class="co">#&gt;  [49] abind_1.4-5             compiler_4.3.3          withr_3.0.0            </span></span>
<span><span class="co">#&gt;  [52] bit64_4.0.5             doParallel_1.0.17       htmlTable_2.4.2        </span></span>
<span><span class="co">#&gt;  [55] backports_1.4.1         BiocParallel_1.36.0     carData_3.0-5          </span></span>
<span><span class="co">#&gt;  [58] DBI_1.2.2               ggsignif_0.6.4          rjson_0.2.21           </span></span>
<span><span class="co">#&gt;  [61] tools_4.3.3             vipor_0.4.7             foreign_0.8-86         </span></span>
<span><span class="co">#&gt;  [64] beeswarm_0.4.0          msigdbr_7.5.1           nnet_7.3-19            </span></span>
<span><span class="co">#&gt;  [67] glue_1.7.0              grid_4.3.3              checkmate_2.3.1        </span></span>
<span><span class="co">#&gt;  [70] cluster_2.1.6           fgsea_1.28.0            generics_0.1.3         </span></span>
<span><span class="co">#&gt;  [73] gtable_0.3.4            preprocessCore_1.64.0   data.table_1.15.2      </span></span>
<span><span class="co">#&gt;  [76] WGCNA_1.72-5            car_3.1-2               utf8_1.2.4             </span></span>
<span><span class="co">#&gt;  [79] XVector_0.42.0          foreach_1.5.2           pillar_1.9.0           </span></span>
<span><span class="co">#&gt;  [82] stringr_1.5.1           babelgene_22.9          limma_3.58.1           </span></span>
<span><span class="co">#&gt;  [85] circlize_0.4.16         splines_4.3.3           BiocFileCache_2.10.1   </span></span>
<span><span class="co">#&gt;  [88] lattice_0.22-5          survival_3.5-8          bit_4.0.5              </span></span>
<span><span class="co">#&gt;  [91] tidyselect_1.2.0        GO.db_3.18.0            ComplexHeatmap_2.18.0  </span></span>
<span><span class="co">#&gt;  [94] locfit_1.5-9.9          Biostrings_2.70.2       knitr_1.45             </span></span>
<span><span class="co">#&gt;  [97] gridExtra_2.3           IRanges_2.36.0          edgeR_4.0.16           </span></span>
<span><span class="co">#&gt; [100] stats4_4.3.3            xfun_0.42               statmod_1.5.0          </span></span>
<span><span class="co">#&gt; [103] matrixStats_1.2.0       stringi_1.8.3           yaml_2.3.8             </span></span>
<span><span class="co">#&gt; [106] evaluate_0.23           codetools_0.2-19        cli_3.6.2              </span></span>
<span><span class="co">#&gt; [109] ontologyIndex_2.12      rpart_4.1.23            systemfonts_1.0.5      </span></span>
<span><span class="co">#&gt; [112] munsell_0.5.0           jquerylib_0.1.4         Rcpp_1.0.12            </span></span>
<span><span class="co">#&gt; [115] GenomeInfoDb_1.38.6     dbplyr_2.4.0            png_0.1-8              </span></span>
<span><span class="co">#&gt; [118] fastcluster_1.2.6       parallel_4.3.3          pkgdown_2.0.7          </span></span>
<span><span class="co">#&gt; [121] ggplot2_3.5.0           blob_1.2.4              bitops_1.0-7           </span></span>
<span><span class="co">#&gt; [124] scales_1.3.0            crayon_1.5.2            GetoptLong_1.0.5       </span></span>
<span><span class="co">#&gt; [127] rlang_1.1.3             cowplot_1.1.3           fastmatch_1.1-4        </span></span>
<span><span class="co">#&gt; [130] KEGGREST_1.42.0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0" line-spacing="2">
<div id="ref-korotkevich_fgsea_2019" class="csl-entry">
1. Korotkevich, G., Sukhov, V., &amp; Sergushichev, A. (2019). Fast gene
set enrichment analysis. <em>bioRxiv</em>. <a href="https://doi.org/10.1101/060012" class="external-link">https://doi.org/10.1101/060012</a>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Tyler J. Sagendorf, Zhenxin Hou.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
