<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>WGCNA of Omics Datasets â€¢ MotrpacRatTraining6moWATData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="WGCNA of Omics Datasets">
<meta property="og:description" content="MotrpacRatTraining6moWATData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MotrpacRatTraining6moWATData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ADIPOCYTE_SIZE_STATS.html">Statistical analysis of adipocyte diameter</a>
    </li>
    <li>
      <a href="../articles/ANALYTES_STATS.html">Statistical analyses of plasma clinical analytes</a>
    </li>
    <li>
      <a href="../articles/FGSEA_heatmaps.html">Heatmaps of FGSEA Results</a>
    </li>
    <li>
      <a href="../articles/KSEA_heatmaps.html">Heatmaps of Kinase-Substrate Enrichment Analysis Results</a>
    </li>
    <li>
      <a href="../articles/limma_diagnostics.html">LIMMA Diagnostic Plots</a>
    </li>
    <li>
      <a href="../articles/lipid_regulatory_proteins_heatmap.html">Heatmap of lipid regulatory proteins</a>
    </li>
    <li>
      <a href="../articles/MAIN_FGSEA.html">FGSEA of Omics Datasets</a>
    </li>
    <li>
      <a href="../articles/METAB_TAG_heatmap.html">Heatmap of total triacylglycerol concentrations</a>
    </li>
    <li>
      <a href="../articles/MitoCarta_FGSEA_heatmaps.html">Plots of MitoCarta 3.0 FGSEA Results</a>
    </li>
    <li>
      <a href="../articles/PHENO_WAT_STATS.html">Statistical analyses of post - pre training differences</a>
    </li>
    <li>
      <a href="../articles/PHOSPHO_KSEA.html">Kinase-Substrate Enrichment Analysis</a>
    </li>
    <li>
      <a href="../articles/plot_adipocyte_size_distribution.html">Plots of Adipocyte Size Distributions</a>
    </li>
    <li>
      <a href="../articles/plot_analytes.html">Plots of plasma clinical analytes</a>
    </li>
    <li>
      <a href="../articles/plot_FGSEA_male_vs_female_summary.html">Plots of Top FGSEA Terms from Male vs. Female Comparisons</a>
    </li>
    <li>
      <a href="../articles/plot_Hk2.html">Plot of hexokinase 2 (Hk2) protein log2 relative abundance</a>
    </li>
    <li>
      <a href="../articles/plot_mito_reads_and_cardiolipin.html">Plots of % mitochondrial reads and cardiolipin</a>
    </li>
    <li>
      <a href="../articles/plot_mtDNA.html">Plot of Mitochondrial DNA (mtDNA)</a>
    </li>
    <li>
      <a href="../articles/plot_pct_fat_vs_leptin.html">Plot of Change in % Body Fat vs. Plasma Leptin</a>
    </li>
    <li>
      <a href="../articles/plot_pre_post.html">Plots of pre and post training values</a>
    </li>
    <li>
      <a href="../articles/plot_proteins.html">Plots of log2 relative abundances of various proteins</a>
    </li>
    <li>
      <a href="../articles/plot_TAG_chain_length_vs_logFC.html">Plots of TAG chain length or number of double bonds vs. log2 fold-change</a>
    </li>
    <li>
      <a href="../articles/plot_UpSet.html">Differential analysis UpSet Plots</a>
    </li>
    <li>
      <a href="../articles/PROT_MITOCARTA_FGSEA.html">Proteomics MitoCarta FGSEA</a>
    </li>
    <li>
      <a href="../articles/volcano_plots.html">Volcano Plots</a>
    </li>
    <li>
      <a href="../articles/WAT_DA.html">Differential Analysis of Omics Datasets</a>
    </li>
    <li>
      <a href="../articles/WAT_WGCNA.html">WGCNA of Omics Datasets</a>
    </li>
    <li>
      <a href="../articles/WGCNA_correlation_heatmaps.html">Heatmaps of WGCNA Module Correlations</a>
    </li>
    <li>
      <a href="../articles/WGCNA_eigenfeature_plots.html">Plots of WGCNA eigenfeatures for each -ome</a>
    </li>
    <li>
      <a href="../articles/WGCNA_METAB_module_heatmaps.html">Heatmaps of Metabolites Grouped by WGCNA Module</a>
    </li>
    <li>
      <a href="../articles/WGCNA_module_ORA.html">Plots of WGCNA Module Over-Representation Analysis Results</a>
    </li>
    <li>
      <a href="../articles/WGCNA_MODULE_ORA.html">WGCNA Module ORA</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moWATData/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>WGCNA of Omics Datasets</h1>
                        <h4 data-toc-skip class="author">Tyler
Sagendorf</h4>
            
            <h4 data-toc-skip class="date">01 May, 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moWATData/blob/HEAD/vignettes/WAT_WGCNA.Rmd" class="external-link"><code>vignettes/WAT_WGCNA.Rmd</code></a></small>
      <div class="hidden name"><code>WAT_WGCNA.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://motrpac.github.io/MotrpacRatTraining6moWATData/" class="external-link">MotrpacRatTraining6moWATData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/" class="external-link">MotrpacRatTraining6moWAT</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">edgeR</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="metabolomics">Metabolomics<a class="anchor" aria-label="anchor" href="#metabolomics"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># This is quick to run</span></span>
<span><span class="va">METAB_WGCNA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/reference/run_WGCNA.html" class="external-link">run_WGCNA</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">METAB_EXP</span>,</span>
<span>                         power <span class="op">=</span> <span class="fl">12</span>,</span>
<span>                         module_prefix <span class="op">=</span> <span class="st">"M"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ..connectivity..</span></span>
<span><span class="co">#&gt; ..matrix multiplication (system BLAS)..</span></span>
<span><span class="co">#&gt; ..normalization..</span></span>
<span><span class="co">#&gt; ..done.</span></span>
<span><span class="co">#&gt;  ..done.</span></span></code></pre></div>
<p><img src="WAT_WGCNA_files/figure-html/unnamed-chunk-2-1.png" width="700"><img src="WAT_WGCNA_files/figure-html/unnamed-chunk-2-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt;  mergeCloseModules: Merging modules whose distance is less than 0.15</span></span>
<span><span class="co">#&gt;    Calculating new MEs...</span></span>
<span><span class="co">#&gt;    multiSetMEs: Calculating module MEs.</span></span>
<span><span class="co">#&gt;      Working on set 1 ...</span></span>
<span><span class="co">#&gt;      moduleEigengenes: Calculating 8 module eigengenes in given set.</span></span>
<span><span class="co">#&gt;       ..principal component calculation for module red failed with the following error:</span></span>
<span><span class="co">#&gt;            Error in impute.knn(datModule, k = min(10, nrow(datModule) - 1)) : </span></span>
<span><span class="co">#&gt;   a column has more than 80 % missing values!</span></span>
<span><span class="co">#&gt;        ..hub genes will be used instead of principal components.</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">METAB_WGCNA</span><span class="op">$</span><span class="va">modules</span><span class="op">$</span><span class="va">moduleID</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  M0  M1  M2  M3  M4  M5  M6  M7 </span></span>
<span><span class="co">#&gt;   6 415 221 137  99  86  69  30</span></span>
<span><span class="co"># M0  M1  M2  M3  M4  M5  M6  M7</span></span>
<span><span class="co">#  6 415 221 137  99  86  69  30</span></span></code></pre>
<p><img src="WAT_WGCNA_files/figure-html/unnamed-chunk-2-3.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="proteomics">Proteomics<a class="anchor" aria-label="anchor" href="#proteomics"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Proportion of missing values?</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/proportions.html" class="external-link">prop.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">PROT_EXP</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="co"># ~5.7% of values are missing</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## NOT RUN WHEN BUILDING VIGNETTE (too slow)</span></span>
<span><span class="va">PROT_WGCNA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/reference/run_WGCNA.html" class="external-link">run_WGCNA</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PROT_EXP</span>,</span>
<span>                        power <span class="op">=</span> <span class="fl">12</span>,</span>
<span>                        module_prefix <span class="op">=</span> <span class="st">"P"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">PROT_WGCNA</span><span class="op">$</span><span class="va">modules</span><span class="op">$</span><span class="va">moduleID</span><span class="op">)</span></span>
<span><span class="co">#   P1   P2   P3   P4   P5   P6   P7   P8   P9  P10  P11</span></span>
<span><span class="co"># 3984 1444 1412  734  696  440  435  235  227  192  165</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="transcriptomics">Transcriptomics<a class="anchor" aria-label="anchor" href="#transcriptomics"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert filtered counts to normalized log2 counts-per-million reads</span></span>
<span><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">TRNSCRPT_EXP</span><span class="op">)</span>,</span>
<span>               samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">TRNSCRPT_EXP</span><span class="op">)</span>,</span>
<span>               group <span class="op">=</span> <span class="va">TRNSCRPT_EXP</span><span class="op">$</span><span class="va">exp_group</span><span class="op">)</span></span>
<span><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></span><span class="op">(</span><span class="va">dge</span>, method <span class="op">=</span> <span class="st">"TMM"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">TRNSCRPT_EXP</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">dge</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## NOT RUN WHEN BUILDING VIGNETTE (too slow: ~1 hr)</span></span>
<span><span class="va">TRNSCRPT_WGCNA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6moWAT/reference/run_WGCNA.html" class="external-link">run_WGCNA</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">TRNSCRPT_EXP</span>,</span>
<span>                            power <span class="op">=</span> <span class="fl">25</span>, <span class="co"># use power = 20:30 to see plots</span></span>
<span>                            module_prefix <span class="op">=</span> <span class="st">"T"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">TRNSCRPT_WGCNA</span><span class="op">$</span><span class="va">modules</span><span class="op">$</span><span class="va">moduleID</span><span class="op">)</span></span>
<span><span class="co">#   T0   T1   T2   T3   T4   T5   T6   T7   T8   T9  T10  T11  T12  T13  T14</span></span>
<span><span class="co"># 2587 4683 3251 2134 1517  541  448  325  226  210  186  114   98   51   33</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Save</span></span>
<span><span class="fu">usethis</span><span class="fu">::</span><span class="fu">use_data</span><span class="op">(</span><span class="va">METAB_WGCNA</span>, internal <span class="op">=</span> <span class="cn">FALSE</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  version <span class="op">=</span> <span class="fl">3</span>, compress <span class="op">=</span> <span class="st">"bzip2"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">usethis</span><span class="fu">::</span><span class="fu">use_data</span><span class="op">(</span><span class="va">PROT_WGCNA</span>, internal <span class="op">=</span> <span class="cn">FALSE</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  version <span class="op">=</span> <span class="fl">3</span>, compress <span class="op">=</span> <span class="st">"bzip2"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">usethis</span><span class="fu">::</span><span class="fu">use_data</span><span class="op">(</span><span class="va">TRNSCRPT_WGCNA</span>, internal <span class="op">=</span> <span class="cn">FALSE</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  version <span class="op">=</span> <span class="fl">3</span>, compress <span class="op">=</span> <span class="st">"bzip2"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] edgeR_4.2.0                        limma_3.60.0                      </span></span>
<span><span class="co">#&gt; [3] MotrpacRatTraining6moWAT_1.0.1     Biobase_2.64.0                    </span></span>
<span><span class="co">#&gt; [5] BiocGenerics_0.50.0                MotrpacRatTraining6moWATData_2.0.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3      rstudioapi_0.16.0       jsonlite_1.8.8         </span></span>
<span><span class="co">#&gt;   [4] shape_1.4.6.1           magrittr_2.0.3          ggbeeswarm_0.7.2       </span></span>
<span><span class="co">#&gt;   [7] rmarkdown_2.26          GlobalOptions_0.1.2     fs_1.6.4               </span></span>
<span><span class="co">#&gt;  [10] zlibbioc_1.50.0         ragg_1.3.0              vctrs_0.6.5            </span></span>
<span><span class="co">#&gt;  [13] memoise_2.0.1           base64enc_0.1-3         rstatix_0.7.2          </span></span>
<span><span class="co">#&gt;  [16] htmltools_0.5.8.1       dynamicTreeCut_1.63-1   curl_5.2.1             </span></span>
<span><span class="co">#&gt;  [19] broom_1.0.5             Formula_1.2-5           sass_0.4.9             </span></span>
<span><span class="co">#&gt;  [22] bslib_0.7.0             htmlwidgets_1.6.4       desc_1.4.3             </span></span>
<span><span class="co">#&gt;  [25] impute_1.78.0           cachem_1.0.8            lifecycle_1.0.4        </span></span>
<span><span class="co">#&gt;  [28] iterators_1.0.14        pkgconfig_2.0.3         Matrix_1.7-0           </span></span>
<span><span class="co">#&gt;  [31] R6_2.5.1                fastmap_1.1.1           GenomeInfoDbData_1.2.12</span></span>
<span><span class="co">#&gt;  [34] clue_0.3-65             digest_0.6.35           colorspace_2.1-0       </span></span>
<span><span class="co">#&gt;  [37] patchwork_1.2.0         AnnotationDbi_1.65.2    S4Vectors_0.42.0       </span></span>
<span><span class="co">#&gt;  [40] textshaping_0.3.7       Hmisc_5.1-2             RSQLite_2.3.6          </span></span>
<span><span class="co">#&gt;  [43] ggpubr_0.6.0            filelock_1.0.3          latex2exp_0.9.6        </span></span>
<span><span class="co">#&gt;  [46] fansi_1.0.6             httr_1.4.7              abind_1.4-5            </span></span>
<span><span class="co">#&gt;  [49] compiler_4.4.0          bit64_4.0.5             doParallel_1.0.17      </span></span>
<span><span class="co">#&gt;  [52] htmlTable_2.4.2         backports_1.4.1         BiocParallel_1.38.0    </span></span>
<span><span class="co">#&gt;  [55] carData_3.0-5           DBI_1.2.2               highr_0.10             </span></span>
<span><span class="co">#&gt;  [58] ggsignif_0.6.4          rjson_0.2.21            tools_4.4.0            </span></span>
<span><span class="co">#&gt;  [61] vipor_0.4.7             foreign_0.8-86          beeswarm_0.4.0         </span></span>
<span><span class="co">#&gt;  [64] msigdbr_7.5.1           nnet_7.3-19             glue_1.7.0             </span></span>
<span><span class="co">#&gt;  [67] grid_4.4.0              checkmate_2.3.1         cluster_2.1.6          </span></span>
<span><span class="co">#&gt;  [70] fgsea_1.30.0            generics_0.1.3          gtable_0.3.5           </span></span>
<span><span class="co">#&gt;  [73] preprocessCore_1.66.0   tidyr_1.3.1             data.table_1.15.4      </span></span>
<span><span class="co">#&gt;  [76] WGCNA_1.72-5            car_3.1-2               utf8_1.2.4             </span></span>
<span><span class="co">#&gt;  [79] XVector_0.44.0          foreach_1.5.2           pillar_1.9.0           </span></span>
<span><span class="co">#&gt;  [82] stringr_1.5.1           babelgene_22.9          circlize_0.4.16        </span></span>
<span><span class="co">#&gt;  [85] splines_4.4.0           dplyr_1.1.4             BiocFileCache_2.12.0   </span></span>
<span><span class="co">#&gt;  [88] lattice_0.22-6          survival_3.5-8          bit_4.0.5              </span></span>
<span><span class="co">#&gt;  [91] tidyselect_1.2.1        GO.db_3.19.1            ComplexHeatmap_2.20.0  </span></span>
<span><span class="co">#&gt;  [94] locfit_1.5-9.9          Biostrings_2.72.0       knitr_1.46             </span></span>
<span><span class="co">#&gt;  [97] gridExtra_2.3           IRanges_2.38.0          stats4_4.4.0           </span></span>
<span><span class="co">#&gt; [100] xfun_0.43               statmod_1.5.0           matrixStats_1.3.0      </span></span>
<span><span class="co">#&gt; [103] stringi_1.8.3           UCSC.utils_1.0.0        yaml_2.3.8             </span></span>
<span><span class="co">#&gt; [106] evaluate_0.23           codetools_0.2-20        tibble_3.2.1           </span></span>
<span><span class="co">#&gt; [109] cli_3.6.2               ontologyIndex_2.12      rpart_4.1.23           </span></span>
<span><span class="co">#&gt; [112] systemfonts_1.0.6       munsell_0.5.1           jquerylib_0.1.4        </span></span>
<span><span class="co">#&gt; [115] Rcpp_1.0.12             GenomeInfoDb_1.40.0     dbplyr_2.5.0           </span></span>
<span><span class="co">#&gt; [118] png_0.1-8               fastcluster_1.2.6       parallel_4.4.0         </span></span>
<span><span class="co">#&gt; [121] pkgdown_2.0.9           ggplot2_3.5.1           blob_1.2.4             </span></span>
<span><span class="co">#&gt; [124] scales_1.3.0            purrr_1.0.2             crayon_1.5.2           </span></span>
<span><span class="co">#&gt; [127] GetoptLong_1.0.5        rlang_1.1.3             cowplot_1.1.3          </span></span>
<span><span class="co">#&gt; [130] fastmatch_1.1-4         KEGGREST_1.44.0</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Tyler J. Sagendorf, Zhenxin Hou.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
